src/app/components/content-viewer/content-viewer.component.ts

```diff
@@ -22,8 +22,16 @@ export class ContentViewerComponent implements OnDestroy {
   contentChange = new BehaviorSubject<GenericElementData>(undefined);
   @ViewChild('container', { read: ViewContainerRef, static: false }) container: ViewContainerRef;
 
+  constructor(
+    private componentRegister: ComponentRegisterService,
+  ) {
+  }
   // tslint:disable-next-line: no-any
   public parsedContent: Observable<{ [keyName: string]: any }> = this.contentChange.pipe(
+    map((data) => ({
+      ...data,
+      type: this.componentRegister.getComponent(data.type),
```

Comment:
Move this to `content-viewer.component` changing  `public parsedContent: Observable ...` like you do inside this `map`

Reply:
It is already in content-viewer.component